File: sdks/php/src/Resources/ExpensesResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\Expense;
use Puxbay\Models\PaginatedResponse;

class ExpensesResource extends BaseResource
{
    public function list(int $page = 1, ?string $category = null): PaginatedResponse
    {
        $endpoint = "expenses/?page={$page}";
        if ($category) {
            $endpoint .= "&category={$category}";
        }
        $data = $this->get($endpoint);
        return PaginatedResponse::fromArray($data, fn($item) => Expense::fromArray($item));
    }

    public function getExpense(string $expenseId): Expense
    {
        $data = $this->get("expenses/{$expenseId}/");
        return Expense::fromArray($data);
    }

    public function create(Expense $expense): Expense
    {
        $data = $this->post('expenses/', $expense->toArray());
        return Expense::fromArray($data);
    }

    public function update(string $expenseId, Expense $expense): Expense
    {
        $data = $this->patch("expenses/{$expenseId}/", $expense->toArray());
        return Expense::fromArray($data);
    }

    public function deleteExpense(string $expenseId): void
    {
        $this->delete("expenses/{$expenseId}/");
    }

    public function listCategories(): array
    {
        // Simple list of categories, keeping as array
        return $this->get("expense-categories/");
    }
}

File: sdks/dotnet/src/Resources/ExpensesResource.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class ExpensesResource
    {
        private readonly PuxbayClient _client;
        public ExpensesResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<Expense>> ListAsync(int page = 1, string category = null)
        {
            string query = $"expenses/?page={page}";
            if (!string.IsNullOrEmpty(category)) query += $"&category={category}";
            return await _client.GetAsync<PaginatedResponse<Expense>>(query);
        }

        public async Task<Expense> GetAsync(string expenseId)
        {
            return await _client.GetAsync<Expense>($"expenses/{expenseId}/");
        }

        public async Task<Expense> CreateAsync(Expense expense)
        {
            return await _client.PostAsync<Expense>("expenses/", expense);
        }

        public async Task<Expense> UpdateAsync(string expenseId, Expense expense)
        {
            return await _client.PatchAsync<Expense>($"expenses/{expenseId}/", expense);
        }

        public async Task DeleteAsync(string expenseId)
        {
            await _client.DeleteAsync($"expenses/{expenseId}/");
        }
        
        public async Task<List<string>> ListCategoriesAsync()
        {
            return await _client.GetAsync<List<string>>("expense-categories/");
        }
    }
}
